File: src/main/java/dev/tricht/lunaris/util/KeyboardUtil.java
package dev.tricht.lunaris.util;

import lombok.extern.slf4j.Slf4j;

import java.awt.*;
import java.awt.datatransfer.StringSelection;
import java.awt.event.KeyEvent;

@Slf4j
public class KeyboardUtil {

    public static void type(Robot robot, String text) {
        Toolkit.getDefaultToolkit().getSystemClipboard().setContents(new StringSelection(text), null);
        robot.keyPress(KeyEvent.VK_CONTROL);
        robot.keyPress(KeyEvent.VK_V);
        robot.keyRelease(KeyEvent.VK_V);
        robot.keyRelease(KeyEvent.VK_CONTROL);
    }

}

File: src/main/java/dev/tricht/lunaris/util/KeyboardUtil.kt
package dev.tricht.lunaris.util

import java.awt.*
import java.awt.datatransfer.StringSelection
import java.awt.event.KeyEvent

object KeyboardUtil {
    fun type(robot: Robot, text: String) {
        Toolkit.getDefaultToolkit().systemClipboard.setContents(StringSelection(text), null)
        robot.keyPress(KeyEvent.VK_CONTROL)
        robot.keyPress(KeyEvent.VK_V)
        robot.keyRelease(KeyEvent.VK_V)
        robot.keyRelease(KeyEvent.VK_CONTROL)
    }
}
